src/tools/fuzzing.h

```diff
@@ -432,21 +461,83 @@ class TranslateToFuzzReader {
   void setupEvents() {
     Index num = upTo(3);
     for (size_t i = 0; i < num; i++) {
-      auto* event =
-        builder.makeEvent(std::string("event$") + std::to_string(i),
-                          WASM_EVENT_ATTRIBUTE_EXCEPTION,
-                          Signature(getControlFlowType(), Type::none));
-      wasm.addEvent(event);
+      auto name = std::string("event$") + std::to_string(i);
+      if (!wasm.getEventOrNull(name)) {
+        auto* event =
+          builder.makeEvent(name,
+                            WASM_EVENT_ATTRIBUTE_EXCEPTION,
+                            Signature(getControlFlowType(), Type::none));
+        wasm.addEvent(event);
+      }
+    }
+  }
+
+  void finalizeMemory() {
+    for (auto& segment : wasm.memory.segments) {
+      Address maxOffset = segment.data.size();
+      if (!segment.isPassive) {
+        // If the offset is a global that was imported (which is ok) but no
+        // longer is (not ok) we need to change that.
+        if (auto* offset = segment.offset->dynCast<GlobalGet>()) {
+          if (!wasm.getGlobal(offset->name)->imported()) {
```

Comment:
I'll improve the comment to this:
```
        // It is legal to use an imported global as a segment initializer, but
        // *not* to use a non-imported global. We have made all globals be
        // non-imported earlier (to avoid having to handle arbitrary imports
        // in the fuzzer harness), which means we may refer to a global here
        // that used to be imported, but no longer is, which we must fix up.
```

Reply:
Thanks. But isn't `if (!wasm.getGlobal(offset->name)->imported())` always true? Do we need this check?

Reply:
We do make all the globals in the initial contents be non-imported. But there may be other globals that the fuzzer adds, that are imported. I think there aren't any atm, but this code would handle that.

I agree it looks confusing atm, I'll add a comment.